fdbserver/RESTKmsConnector.actor.cpp

```diff
@@ -285,7 +287,23 @@ ACTOR Future<Void> discoverKmsUrls(Reference<RESTKmsConnectorCtx> ctx, RefreshPe
 	return Void();
 }
 
-void checkResponseForError(Reference<RESTKmsConnectorCtx> ctx, const rapidjson::Document& doc) {
+void checkResponseForError(Reference<RESTKmsConnectorCtx> ctx, const rapidjson::Document& doc, bool isCipher) {
+	// check version tag sanity
+	if (!doc.HasMember(REQUEST_VERSION_TAG) || !doc[REQUEST_VERSION_TAG].IsInt()) {
+		TraceEvent("KMSResponseMissingVersion", ctx->uid).log();
+		throw operation_failed();
+	} else {
+		const int version = doc[REQUEST_VERSION_TAG].GetInt();
+		const int maxSupportedVersion = isCipher ? SERVER_KNOBS->REST_KMS_CURRENT_CIPHER_REQUEST_VERSION
+		                                         : SERVER_KNOBS->REST_KMS_CURRENT_BLOB_METADATA_REQUEST_VERSION;
+		if (version == INVALID_REQUEST_VERSION || version > maxSupportedVersion) {
+			TraceEvent("KMSResponseInvalidVersion", ctx->uid)
```

Comment:
Do we want to make this a SevWarn?

Reply:
Agreed, will also update the TraceEvents in this routine as they all should share the same level of logging.